spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Utils.scala

```diff
@@ -0,0 +1,212 @@
+ * limitations under the License.
+ */
+
+
+package com.intel.analytics.bigdl.models.maskrcnn
+
+import breeze.linalg.{*, dim, max}
+import com.intel.analytics.bigdl.nn.ResizeBilinear
+import com.intel.analytics.bigdl.nn.abstractnn.DataFormat
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.sun.xml.internal.bind.v2.TODO
+
+import scala.collection.mutable.ArrayBuffer
+
+object Utils {
+  // box with 4 element (xyxy)
+  def expandBoxes(bbox: Tensor[Float], bboxExpand: Tensor[Float], scale: Float)
+  : Unit = {
+    require(bbox.nElement() == 4 && bboxExpand.nElement() == 4
+      && bbox.dim() == 1 && bboxExpand.dim() == 1,
+      "Box and expected Box should have 4 elements with one dim")
+
+    val box0 = bbox.valueAt(1)
+    val box1 = bbox.valueAt(2)
+    val box2 = bbox.valueAt(3)
+    val box3 = bbox.valueAt(4)
+
+    var wHalf = (box2 - box0) * 0.5f
+    var hHalf = (box3  - box1) * 0.5f
+    val x_c = (box2 + box0) * 0.5f
+    val y_c = (box3 + box1) * 0.5f
+
+    wHalf *= scale
+    hHalf *= scale
+
+    bboxExpand.setValue(1, x_c - wHalf)
+    bboxExpand.setValue(3, x_c + wHalf)
+    bboxExpand.setValue(2, y_c - hHalf)
+    bboxExpand.setValue(4, y_c + hHalf)
+  }
+
+  // mask with three dims (channel, height, wide)
+  def expandMasks(mask: Tensor[Float], padding: Int): (Tensor[Float], Float) = {
+    require(mask.isContiguous(), "Only support contiguous mask")
+
+    val N = mask.size(1)
+    val M = mask.size(mask.dim() - 1)
+    val pad2 = 2 * padding
+    val scale = (M + pad2).toFloat / M
+    val paddedMask = Tensor[Float](N, M + pad2, M + pad2)
+
+    val maskHeight = mask.size(2)
+    val maskWidth = mask.size(3)
+    val padHeight = paddedMask.size(2)
+    val padWidth = paddedMask.size(3)
+
+    for (i <- 1 to  N) {
+      val maskPart = mask.select(1, i)
+      val maskArray = maskPart.storage().array()
+      val maskOffset = maskPart.storageOffset() - 1
+
+      val padPart = paddedMask.select(1, i)
+      val padArray = padPart.storage().array()
+      val padOffset = padPart.storageOffset() - 1
+
+      val nElement = padPart.nElement()
+      for (j <- 0 until nElement) {
+        val tempHeight = j / padWidth + 1
+        val tempWidth = j % padWidth + 1
+        val tempMaskHeight =
+          if ((tempHeight > padding + maskHeight) || (tempHeight < padding)) -1
+          else tempHeight - padding
+
+        val tempMaskWidth =
+          if ((tempWidth > padding + maskWidth) || (tempWidth < padding)) -1
+          else tempWidth - padding
+
+        if (tempMaskHeight > 0 && tempMaskWidth > 0) {
+          val offset = (tempMaskHeight - 1) * maskWidth + tempMaskWidth - 1
+          padArray(j + padOffset) = maskArray(offset + maskOffset)
+        }
+      }
+    }
+    (paddedMask, scale)
+  }
+
+  // mask and box should be one by one
+  def decodeMaskInImage(mask: Tensor[Float], box: Tensor[Float],
+    thresh: Float = 0.5f, padding : Int = 1, binaryMask: Tensor[Float]): Unit = {
```

Comment:
put the parameters with default value to last ones ?

Reply:
ok,changed

Reply:
ok,changed